### ISSUE TITLE
Issue detail screen: enhancement

### ISSUE THREAD

**Issue by MattiaPrimavera** [slapperwan/gh4a#710]
Wouldn't it be comfortable to have a button leading to **repository** home page, when looking at an issue? 

I often have the need to jump from issue to code ...

**Comment by Tunous** [comment]
> I often have the need to jump from issue to code ...

Same. I recently thought about a different solution though. This looks like a good issue to describe it. 

___

The idea is that when you open issue details it launches as a separate task in the recents screen. You could then switch back from it and navigate to other screens like files while keeping your issue separate. Additional gain would be possibility to use Android split screen feature to view issue and files, etc at the same time.

**Comment by smichel17** [comment]
> The idea is that when you open issue details it launches as a separate task in the recents screen.

One task per issue, or one task for the most recently opened issue?

**Comment by Tunous** [comment]
One task per issue so you could split view and see 2 different issues at the same time.

**Comment by Tunous** [comment]
![Demo](https://thumbs.gfycat.com/SecondCavernousLacewing-size_restricted.gif)

Took me few hours and hundreds of stackowerflow/documentation pages (I had problem with making it so single issue wouldn't get duplicated) but I've managed to implement a working solution. It's not finished yet but I must say that I'm **really excited** about this feature. It already works flawlessly and solves many issues that I had with the app while commenting (searching issues to reference, searching code/files to mention, and more)

Only issue right now is that it can no longer be used with startActivityForResult but perhaps there are alternative solutions to that. 

I'll create a pull request later once everything is finished. Just wanted to share how it looks like.

**Comment by maniac103** [comment]
I wonder whether leaving the issue details in the same task by default and having a 'detach' button would be worth a thought?

**Comment by MattiaPrimavera** [comment]
@Tunous Why that can no longer be used used with **startActivityForResult**?

**Comment by Tunous** [comment]
> Why that can no longer be used used with startActivityForResult?

The activity can no longer communicate with the activity that launched it because it is now in a different task. Result communication works only if they are in the same task.

From [startActivityForResult](https://developer.android.com/reference/android/app/Activity.html#startActivityForResult%28android.content.Intent,%20int%29):
> Note that this method should only be used with Intent protocols that are defined to return a result. In other protocols (such as ACTION_MAIN or ACTION_VIEW), you may not get the result when you expect. For example, if the activity you are launching uses FLAG_ACTIVITY_NEW_TASK, it will not run in your task and thus you will immediately receive a cancel result.

_I'm not using FLAG_ACTIVITY_NEW_TASK_ but the same applies. 

> if another task is added to the mix (e.g. you need to call somebody) and the back button doesn't lead you back to the issue list

With current behavior it also might not lead you back to the issue list (launched from link). For that there is up button which should behave as expected also with new behavior.

**Comment by Tunous** [comment]
BTW, Android displays different animation when activity is displayed in new task so it's not that surprising. It's easy to get what it means once you see it (and the recents screen) few times.

**Comment by smichel17** [comment]
The only other app that works like this that I've used is chrome with its tabs, and personally I don't like it clogging up the recents screen (in large part this is Android's fault for having no notion of subtasks (eg, pick your app, then your task from it, or more importantly, swipe away all tasks from the same app at once).

I'd prefer an action to pop an issue out into a new task, or limiting to 2 tasks (one for the main app, one for the last opened issue).

**Comment by Tunous** [comment]
Take a look at [Google Sheets](https://play.google.com/store/apps/details?id=com.google.android.apps.docs.editors.sheets) or [Google Docs](https://play.google.com/store/apps/details?id=com.google.android.apps.docs.editors.docs).
